jena-shacl/src/main/java/org/apache/jena/shacl/parser/ConstraintVisitor.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jena.shacl.parser;
+
+import org.apache.jena.shacl.engine.constraint.*;
+
+public interface ConstraintVisitor {
+    default void visit(ReportConstraint constraint){}
```

Comment:
This one can be skipped.  It's a dummy that only exists to carry a message. It can't be an executable constraint.

Reply:
We must implement `ReportConstraint.visit(ConstraintVisitor)`, though. Empty implementation for minimal friction?

Reply:
That'll do fine. The validation operations are `throw new UnsupportedOperationException()` but no-op works just as well for the visitor. (I suspect that ReportConstraint shouldn't exist at all! But from memory, passing the information it carries means adding other parameters in many places which even if purer, is verbose.)